src/main/jbake/content/blog/2016/10/24/hawkfx-alerts.adoc

```diff
@@ -0,0 +1,99 @@
+= Define Alert Triggers via HawkFX
+Heiko W. Rupp
+2016-10-24
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, ruby, jruby, javafx, alerts
+
+HawkFX is an explorer for Hawkular. It is a pet project of mine and https://github.com/pilhuhn/hawkfx[hosted on
+Github]. It now got the ability to define alert triggers.
+
+Hawkular-services and now also Hawkular-Metrics standalone have the ability to fire alerts when certain conditions
+are met. To define those conditions, one creates a _trigger_ via
+the http://www.hawkular.org/docs/rest/rest-alerts.html[Hawkular-Alerts api]. The alerts team has also written a
+http://www.hawkular.org/community/docs/developer-guide/alerts.html[developer guide] that explains the concepts and
+terms behind Hawkular-Alerts.
+
+HawkFX now has the ability to directly insert new triggers. For this purpose I have created a
+http://pilhuhn.blogspot.de/2016/10/a-dsl-for-alert-trigger-definitions-in.html[DSL] to allow to define the triggers
+in a more 'natural language'. More on this below.
+
+[[img-main]]
+.Context menu
+ifndef::env-github[]
+image::/img/blog/2016/hawkfx-trigger-menu.png[HawkFX trigger context menu,300]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/hawkfx-trigger-menu.png[HawkFx trigger context menu,300]
+endif::[]
+
+
+
+When you right-click on a metric in the HawkFX list of metrics, there is now a new context menu item 'New trigger',
+that leads you to the insert screen, where you can insert the definition. This will contain a pre-canned definition
+to start with.
+
+[[img-main]]
+.Alert trigger insert screen
+ifndef::env-github[]
+image::/img/blog/2016/hawkfx-trigger-insert-box.png[HawkFX trigger insert,600]
+endif::[]
+ifdef::env-github[]
+image::../../../../../assets/img/blog/2016/hawkfx-trigger-insert-box.png[HawkFx trigger insert,600]
+endif::[]
+
+When you are done writing the definition, you can click on the _parse_ button at the bottom to check the definition.
+
+== Examples
+
+.Simple threshold trigger for gaguges
+----
+define trigger "MyTrigger"
+ ( threshold "myvalue" > 3 )
+----
+
+.and for counters
+----
+define trigger "MyTrigger"
+( threshold counter "myvalue" > 3 )
+----
+
+You can see from the previous two examples that gauges are the default for _threshold_ triggers.
+
+.Trigger for availability, with high severity that is initially disabled
+----
+define trigger "MyTrigger"
+  disabled
+  severity HIGH
+  ( availability "mymetric" is DOWN )
+----
+
+.Trigger that fires when two conditions are true
+----
+define trigger "MyTrigger"
+  AND(
+    ( threshold counter "mycount" < 5 )
+    ( string "mymetric" CO "ERROR" )
```

Comment:
CO is what, contains?  If you want natural language I think perhaps just use the full word for the string operation.